File: 012.jarakirefactor/djeiva/oo/objetos/ex1/Principal1.erl
-module('Principal1').
-export([main/1]).

main(_V_args) ->
	Scope = {main, 'Principal1'},

	%% Pessoa p = new Pessoa()
	st:put({Scope, "b"},
		{'Bola',
			oo:new('Bola', [],
				[{"cor", 'String', undefined},
				{"circunferencia", float, 0.0},
				{"material", 'String', undefined}
				])}),

	%% b.trocaCor("azul"); b.mostraCor(); b.trocaCor("preto"); b.mostraCor();
	begin
	Temp_Class1 = oo:get_class(st:get(Scope, "b")),
	Temp_Class1:trocaCor(st:get(Scope, "b"), "azul")
	end,

	begin
	Temp_Class2 = oo:get_class(st:get(Scope, "b")),
	Temp_Class2:mostraCor(st:get(Scope, "b"))
	end,

	begin
	Temp_Class1 = oo:get_class(st:get(Scope, "b")),
	Temp_Class1:trocaCor(st:get(Scope, "b"), "preto")
	end,

	begin
	Temp_Class2 = oo:get_class(st:get(Scope, "b")),
	Temp_Class2:mostraCor(st:get(Scope, "b"))
	end.

File: java_src/oo/objetos/obj4/Principal1.java
public class Principal1 {
	public static void main(String args[]){
		Bola b = new Bola();

		b.c = 10.0;

		System.out.println("Teste circunferencia: " + b.c);

		b.circ(1.0);
		System.out.println("Teste circunferencia2: " + b.c);

		b.trocaCor1("rosa");
		b.mostraCor();

		b.trocaCor2("azul");
		b.mostraCor();

		b.trocaCor2("preto");
		b.mostraCor();


		Bola b2 = new Bola(5.0, "opaaasa");
		System.out.println("circunferencia:  " + b2.c);
		System.out.println("cor: " + b2.cor);
	}
}